src/ebonite/ext/prebuild.py

```diff
@@ -0,0 +1,62 @@
+import os
+import re
+import shutil
+import sys
+import tempfile
+
+from jinja2 import Environment, FileSystemLoader
+
+from ebonite.build.docker import (RemoteDockerRegistry, create_docker_client, image_exists_at_dockerhub,
+                                  login_to_registry, repository_tags_at_dockerhub)
+from ebonite.utils.log import logger
+
+
+def prebuild_image(prebuild_path, name_template, python_version, *, push=False):
+    tag = name_template.format(python_version)
+    if image_exists_at_dockerhub(tag):
+        logger.info('Skipped building image %s: already exists', tag)
+        return
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        context_dir = os.path.join(tmpdir, 'context')
+        logger.info('Building image %s on %s ...', tag, context_dir)
+
+        shutil.copytree(prebuild_path, context_dir)
+        _generate_dockerfile(context_dir, python_version)
+
+        try:
+            with create_docker_client() as client:
+                client.images.build(path=context_dir, tag=tag, rm=True)
+
+            logger.info('Successfully built image %s', tag)
+        except Exception as e:
+            logger.error('Failed to build image %s: %s', tag, e)
+            return
+
+    if push:
+        with create_docker_client() as client:
+            login_to_registry(client, RemoteDockerRegistry())
+            client.images.push(tag)
+
+
+def _generate_dockerfile(context_dir, python_version):
+    j2 = Environment(loader=FileSystemLoader([context_dir]))
+    docker_tmpl = j2.get_template('Dockerfile.j2')
+    with open(os.path.join(context_dir, 'Dockerfile'), 'w') as df:
+        df.write(docker_tmpl.render({'python_version': python_version}))
+
+
+PY_BLACK_LIST = {'^2\\.', '^3\\.[0-5]\\.', '^3\\.6\\.[0-4]'}
```

Comment:
what is up with 3.6.0-3.6.4?

Reply:
Unfortunately the latest uwsgi could not be installed for these versions of Python. That looks strange for me as it works fine for some (not all) 3.5.x versions but that's how it is. I think it should not be a problem as we should start to support 3.8 soon and give up on 3.6 concurrently.